### PR TITLE
Support plugins

### PR DESCRIPTION
**Pull request by jinwoo**
Call initializePlugins before running typecheck.

### CODE REVIEW

**Comment by jinwoo** [comment]
I was going to ask about tests :) Any pointers to how to test this?

File: src/Development/IDE/Core/Compile.hs [hunk 1]
```diff
@@ -95,8 +99,17 @@ typecheckModule (IdeDefer defer) packageState deps pm =
     runGhcEnv packageState $
         catchSrcErrors "typecheck" $ do
             setupEnv deps
+#if MIN_GHC_API_VERSION(8,6,0)
```
Comment (ndmitchell) [inline src/Development/IDE/Core/Compile.hs]: Any way to abstract the key steps out into a function with a single signature that is the same on all versions (merely a noop on one version), then have this code just call that? Would make managing the CPP a bit easier.

**Comment by jinwoo** [comment]
Thanks, @ndmitchell . I'm working on tests at the moment by adding some 3rd-party plugin. Will update this once it's done.

**Comment by jinwoo** [comment]
Added a test. PTAL.

**Comment by cocreature** [comment]
Nice work, thank you!

**Pull request merged**